homeassistant/components/xiaomi_miio/sensor.py

```diff
@@ -145,21 +195,129 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
+                        config_entry,
+                        f"{attribute}_{config_entry.unique_id}",
+                        attribute,
+                    )
+                )
+        else:
+            unique_id = config_entry.unique_id
+            name = config_entry.title
+            _LOGGER.debug("Initializing with host %s (token %s...)", host, token[:5])
 
-        device = AirQualityMonitor(host, token)
-        entities.append(XiaomiAirQualityMonitor(name, device, config_entry, unique_id))
+            device = AirQualityMonitor(host, token)
+            entities.append(
+                XiaomiAirQualityMonitor(name, device, config_entry, unique_id)
+            )
 
     async_add_entities(entities, update_before_add=True)
 
 
+class XiaomiHumidifierSensor(XiaomiMiioEntity, SensorEntity):
+    """Representation of a Xiaomi Humidifier sensor."""
+
+    def __init__(self, name, device, entry, unique_id, attribute):
+        """Initialize the entity."""
+        super().__init__(name, device, entry, unique_id)
+
+        self._name = name
+        self._icon = HUMIDIFIER_SENSOR_TYPES[attribute].icon
+        self._unit_of_measurement = HUMIDIFIER_SENSOR_TYPES[attribute].unit
+        self._device_class = HUMIDIFIER_SENSOR_TYPES[attribute].device_class
+        self._state_class = HUMIDIFIER_SENSOR_TYPES[attribute].state_class
+        self._available = None
+        self._state = None
+        self._state_attrs = {}
+        self._skip_update = False
+        self._attribute = attribute
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity, if any."""
+        return self._unit_of_measurement
+
+    @property
+    def icon(self):
+        """Return the icon to use for device if any."""
+        return self._icon
+
+    @property
+    def available(self):
+        """Return true when state is known."""
+        return self._available
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    @property
+    def extra_state_attributes(self):
+        """Return the state attributes of the device."""
+        return self._state_attrs
+
+    @staticmethod
+    def _extract_value_from_attribute(state, attribute):
+        value = getattr(state, attribute)
+        if isinstance(value, Enum):
+            return value.value
+
+        return value
+
+    async def _try_command(self, mask_error, func, *args, **kwargs):
+        """Call a miio device command handling error messages."""
+        try:
+            result = await self.hass.async_add_executor_job(
+                partial(func, *args, **kwargs)
+            )
+
+            _LOGGER.debug("Response received from miio device: %s", result)
+
+            return result == SUCCESS
+        except DeviceException as exc:
+            if self._available:
+                _LOGGER.error(mask_error, exc)
+                self._available = False
+
+            return False
```

Comment:
This looks like copy&paste from above. I think this is shared among all entities(?), so this should rather belong to `XiaomiMiioEntity` to avoid code duplication.